src/main/java/de/mw/devschool/wordcount/App.java

```diff
@@ -25,8 +27,19 @@ public static void main(String[] args) {
         WordCounter wordCounter = new WordCounter(initializeStopWordsFile());
         final App app = new App(System.out, new BufferedReader(new InputStreamReader(System.in)), wordCounter);
         System.out.println(app.getGreeting());
-
-        app.run();
+        if (args.length == 1) {
+            Path fileName = Path.of(args[0]);
+            try {
+                String text = Files.readString(fileName);
+                long wordCount = wordCounter.countWords(text);
+                app.printWordCount(wordCount);
```

Comment:
I'd prefere it if the app was started with the same method call in all scenarios. You could read the input file in `main` and pass the text to `run`, you could pass the text to the constructor of `App`, or you could even overload the constructor such that you have either the `BufferedReader` _or_ the prepared input text as parameter.